common/src/main/java/io/micrometer/prometheus/MutatedMicrometerCollector.java

```diff
@@ -46,7 +46,7 @@ static TagsHolder from(List<Tag> tags) {
             List<String> values = new ArrayList<>(tags.size());
 
             for (Tag tag : tags) {
-                keys.add(NamingConvention.snakeCase.tagKey(tag.getKey()));
+                keys.add(NamingConvention.snakeCase.tagKey(tag.getKey()).replaceAll("-", "_"));
```

Comment:
NIT:  Should add a comment "Remove post spring 2.7" :) Just so we know this is a workaround due to